var/spack/repos/builtin/packages/py-distributed/package.py

```diff
@@ -77,8 +85,12 @@ class PyDistributed(PythonPackage):
     depends_on("py-tornado@6.0.3:", type=("build", "run"), when="^python@3.8:")
     depends_on("py-tornado@6.0.3:6.1", type=("build", "run"), when="@2022.10.2:")
     depends_on("py-zict@0.1.3:", type=("build", "run"))
+    depends_on("py-zict@2.2.0:", type=("build", "run"), when="@2023.4.1:")
     depends_on("py-pyyaml", type=("build", "run"))
+    depends_on("py-pyyaml@5.3.1:", type=("build", "run"), when="@2023.4.1:")
     depends_on("py-urllib3", type=("build", "run"), when="@2022.10.2:")
+    depends_on("py-urllib3@1.24.3:", type=("build", "run"), when="@2023.4.1:")
```

Comment:
Would be easier to review if these were in the same order as in pyproject.toml and sorted from newest to oldest. Not required though.